alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -78,16 +78,16 @@ void RunnableObject::start(RunningPlan* rp)
         _engine->editScheduler().schedule(initCall);
     } else {
 
-        std::shared_ptr<Blackboard> parentBlackboard;
+        BasicPlan* parentPlan = nullptr;
         if (rp->getParent() && rp->getParent()->getBasicPlan()) {
-            parentBlackboard = rp->getParent()->getBasicPlan()->getBlackboard();
+            parentPlan = rp->getParent()->getBasicPlan();
         }
```

Comment:
`BasicPlan* parentPlan = rp->getParent() ? rp->getParent()->getBasicPlan() : nullptr;`
You can avoid the double call to `getBasicPlan`